tools/base/fpgad/fpgad.c

```diff
@@ -101,6 +109,103 @@ void sig_handler(int sig, siginfo_t *info, void *unused)
 	}
 }
 
+void resolve_dirs(struct config *c)
+{
+	char *sub;
+	bool def;
+	mode_t mode;
+	struct stat stat_buf;
+
+	if (!geteuid()) {
+		// If we're being run as root, then use DEFAULT_DIR_ROOT
+		// as the working directory.
+		strncpy_s(c->directory, sizeof(c->directory),
+				DEFAULT_DIR_ROOT, DEFAULT_DIR_ROOT_SIZE);
+		mode = 0755;
+		c->filemode = 0026;
+	} else {
+		// We're not root. Try to use ${HOME}/.opae
+		char *home = getenv("HOME");
+
+		if (home) {
+			snprintf_s_s(c->directory, sizeof(c->directory),
+					"%s/.opae", home);
+		} else {
+			char cwd[PATH_MAX];
+			// ${HOME} not found - use current dir.
+			if (getcwd(cwd, sizeof(cwd))) {
+				snprintf_s_s(c->directory, sizeof(c->directory),
+						"%s/.opae", cwd);
+			} else {
+				// Current directory not found - use /
+				strncpy_s(c->directory, sizeof(c->directory),
+						"/.opae", 6);
```

Comment:
Does a non-root user have write permission to "/" ?

Reply:
Probably not, however if the user doesn't have a ${HOME}, chances are it's a privileged / system user.  Worst case, the fopen call will fail.  Would it be better to default to /tmp I wonder?

Reply:
I guess no need to sweat on this corner case. It should be fine to default to "/".